<?xml version="1.0"?>
<launch>

<!--     <node pkg="topic_tools" type="throttle" name="csi_throttler"
            args="messages /csi 1 /csi_throttled" /> -->

    <!-- Bearing computation node -->
    <node pkg="bearing_sensor" type="aoa_node.py" name="aoa_node" output="screen" clear_params="true"
            required="true">

        <!-- <remap from="csi" to="csi_throttled" /> -->

        <!-- Code control parameters -->
        <param name="publish_profile"     type="bool"     value="True" /> <!-- Publish profile as image -->
        <param name="publish_channel"     type="bool"     value="True" /> <!-- Publish profile as image -->
        <param name="color_profile"       type="bool"     value="True" /> <!-- Color the profile -->
        <param name="profile_tx_id"       type="int"     value="0" /> <!-- Profile of which transmitter antenna? -->
        <param name="channel_tx_id"       type="int"     value="0" /> <!-- Profile of which transmitter antenna? -->

        <!-- AoA/AoD computation parameters -->
        <param name="algo"                type="string"   value="fft" /> <!-- See attached README for more information -->
        <!-- NUM_THETA_STEPS linearly spaced theta values will be created 
        between THETA_THRESH[0] and THETA_THRESH[1]. Similarly for distance D -->
        <param name="num_theta_steps"     type="int"      value="180" /> 
        <param name="num_d_steps"         type="int"      value="180" />

        <!-- Number of indepedent channels to average over, each channel from each Tx is considered an independent channel -->
        <param name="smoothing_window"    type="int"      value="1" />
        
        <rosparam param="theta_thresh"> [-90, 90] </rosparam>
        <rosparam param="d_thresh"> [-100, 100] </rosparam>

        <!-- Antenna array parameters -->
        <param name="array_type"          type="string"   value="linear" /> <!-- Antenna array shape: square, linear, custom -->
        <param name="num_antennas"        type="int"      value="4" /> <!-- Number of antennas for "linear" array shape; ignored for others -->
        <param name="antenna_sep"         type="double"   value="0.026" /> <!-- Antenna separation, required for "square/linear" array types; ignored otherwise -->
        <rosparam param="rx_position"> [0,  0,
                                        0,  0.052,
                                        0,  0.026,
                                        0,  0.078]
        </rosparam> <!-- Relative locations of all antennas wrt to the first antenna; require for "custom" array types, ignored otherwise -->

        <!-- Filters on the data -->
        <param name="rssi_thresh"         type="double"   value="-100" />
        <rosparam param="valid_tx_ant"> [0] </rosparam>


        <!-- Compensation file path -->
        <param name="comp"                type="string"   value="/home/aarun/Research/data/ros_csi_testing/compensations/comp_224_corrected.npy" />
        <param name="is_conjugated"       type="bool"     value="true" />
        <param name="is_phase_only"       type="bool"     value="true" />
    </node>

    <!-- Image view node -->
    <!-- <node pkg="bearing_sensor" type="runRqt_image_view.sh" name="image_view"  /> -->


</launch>